lib/spack/spack/solver/asp.py

```diff
@@ -528,9 +592,10 @@ def fact(self, head, assumption=False):
             self.assumptions.append(atom)
 
     def solve(
-            self, solver_setup, specs, dump=None, nmodels=0,
-            timers=False, stats=False, tests=False, reuse=False,
+            self, solver_setup, specs_by_psid, dump=None, nmodels=0,
```

Comment:
Not an issue due to this PR, but it seems to me that the `dump` argument is unused in this method (it's used higher in the call stack). I think we can safely remove it.